File: 2685. Count the Number of Complete Components.cpp
class Solution {
    public:
        int countNode = 0;
        vector<bool> visited;
        vector<vector<int>> adj;
        unordered_map<int, int> store;
        void dfs(int node)
        {
            countNode += 1;
            visited[node] = true;
            store[node] = adj[node].size();
            
            for(auto &child : adj[node])
                if(not visited[child]) dfs(child);
        }
        int countCompleteComponents(int n, vector<vector<int>>& edges) 
        {
            adj.resize(n);
            visited.resize(n);
            for(auto &ed : edges)
            {
                adj[ed[0]].push_back(ed[1]);
                adj[ed[1]].push_back(ed[0]);
            }
    
            int ans = 0;
            for(int i = 0; i < n; i++)
            {
                if(not visited[i])
                {
                    dfs(i);
    
                    bool isConnected = true;
                    for(auto [node, count] : store)
                        if(count + 1 != countNode) isConnected = false;
    
                    store.clear();
                    countNode = 0;
                    ans += isConnected;
                }
            }
    
            return ans;
        }
    };

File: 2685. Count the Number of Complete Components.py
class Solution:
    def countCompleteComponents(self, n: int, edges: List[List[int]]) -> int:
        adj = defaultdict(list)

        for u, v in edges:
            adj[u].append(v)
            adj[v].append(u)

        visited = set()

        ans = 0

        for i in range(n):
            if i in visited:
                continue
            visited.add(i)
            
            que = deque([i])
            edj_count, node_count = 0, 0
            while que:
                node = que.popleft()

                node_count += 1
                edj_count += len(adj[node])

                for child in adj[node]:
                    if child in visited:
                        continue
                    visited.add(child)
                    que.append(child)
                    
            ans += (node_count * (node_count - 1)) == edj_count

        return ans